packages/dataviews/src/types.ts

```diff
@@ -158,6 +158,11 @@ export type Field< Item > = {
 	 * Defaults to `item[ field.id ]`.
 	 */
 	getValue?: ( args: { item: Item } ) => any;
+
+	/**
+	 * True if the field is supposed to be used as a media field.
+	 */
+	isMediaField?: boolean;
```

Comment:
I'm not sure this prop should be part of `fields` because it's something used for some specific layouts.

Based on the linked issue's discussions, I got the impression that just rendering the preview would be enough and probably we should try that first.

Having said that, in the future if we want to add more fine control to views that utilize `mediaField`, with `viewConfigOptions` a view could render an extra control for a user to select what to preview in the `mediaField`. It would need some changes on `mediaField` form on that layout, but that is for another time in my opinion.

Reply:
Hi @ntsekouras,

> I'm not sure this prop should be part of fields because it's something used for some specific layouts.

It is still a property of the field, some fields can be media and be rendered as an image or video.
But I'm not sure if isMediaField is a good path either I thought about using a type for media files "type" = "media". Is there any other alternative you would prefer?


> Based on the linked issue's discussions, I got the impression that just rendering the preview would be enough and probably we should try that first.

On https://github.com/WordPress/gutenberg/pull/65331 @jameskoster suggested this design as a UI to switch between media fields. I think some users will prefer a content preview while for others featured-image, so offering the option to choose makes sense. But yah not having the option to choose would be simpler.

@jameskoster do you think not having the field preview switcher in a first version and defaulting to content would be ok?

Reply:
We could still render the control with `viewConfigOptions` prop. 

Having different media fields could be part of the `layout.mediaField` prop, which should be updated in order to support multiple, alternative renders. The latter is also related to the registration of fields, because if a layout provides only one `field` for media field (let's say `featured image`) and the post type doesn't support one, it could be good to have a fallback.

Reply:
> Having different media fields could be part of the layout.mediaField prop

So instead of changing fields at all (not even adding a new type), you would prefer an alternative where layout.mediaField could be an array of fields instead of just one id, and the first item of the array is the media field that is active, correct? I think something like that could also work well.

Reply:
Something like that yes, but haven't explored anything in code, so I'm not sure of any possible nuances. I'll also cc @oandregal and @youknowriad for thoughts.

Reply:
Personally I think it's the right thing for fields to say that they can be used as "media fields". That said, I'm not sure yet what that is. Maybe it's `type: "media"` maybe it's a flag like here but I don't see as part of the "view".

Reply:
Maybe it's even an "interface" for the field. For instance, we could say that "media" fields needs to support a "size" prop in their render function or something...

Reply:
I'd rather introduce the `type: media` field as a way to declare what fields can act as "preview". If we need additional info/props, we can also make that a requirement later.

Reply:
I just checked and the current featuredMedia is of `type: text`, which doesn't make any sense 😓 Tied to making the render more declarative, we may need more granularity: declaring a field is an image, or a video, or an iframe, etc. But perhaps that granularity doesn't need to come with the `type` but with something else that controls the render specifically.